cpp/include/cugraph/graph_functions.hpp

```diff
@@ -412,15 +412,40 @@ decompress_to_edgelist(
 
 /**
  * @ingroup graph_functions_cpp
- * @brief Symmetrize edgelist.
+
+ * @brief Identify if an edge property is part of the uniqueness of the edge, or just a value to be
+ combined.
+ */
+enum class edge_key_selector_t {
+  KEY,  /** This property is part of the uniqueness key of the edge. */
+  VALUE /** This property is a value to be combined. */
+};
+
+/**
+ * @ingroup graph_functions_cpp
+ * @brief Rule for combining edge properties when merging reciprocal edges.
+ */
+enum class edge_value_combining_rule_t {
+  SUM,   /** Sum the values of the two edge properties. */
+  MAX,   /** Take the maximum of the values of the two edge properties. */
+  MIN,   /** Take the minimum of the values of the two edge properties. */
+  MEAN,  /** Take the mean of the values of the two edge properties. */
+  FIRST, /** Use the value of the first edge property. */
+  LAST,  /** Use the value of the last edge property. */
+  ANY,   /** Use the value of any edge property. */
+  NONE   /** Do not combine the values. */
```

Comment:
What is NONE here? Isn't it pretty much identical to edge_key_selector_t::KEY?

Reply:
If set to edge_key_selector::VALUE, I guess we need to combine in one way or another. If we don't want to combine, we need to set to edge_key_selector::KEY.